core/src/main/java/io/ddf/DDFCache.java

```diff
@@ -62,26 +81,83 @@ public void addDDF(DDF ddf) throws DDFException {
     if(ddf.getUUID() == null) {
       throw new DDFException("uuid is null");
     }
-    mDDFCache.put(ddf.getUUID(), ddf);
+    Lock lock = uuidLocks.get(ddf.getUUID());
+    try {
+      if(lock.tryLock(120, TimeUnit.SECONDS)) {
+        try {
+          if (ddf.getName() == null) {
+            mDDFCache.put(ddf.getUUID(), ddf);
+          } else {
+            mDDFwithNameCache.put(ddf.getUUID(), ddf);
+          }
+        } finally {
+          lock.unlock();
+        }
+      } else {
+        throw new DDFException("Timeout getting DDF");
+      }
+    } catch (InterruptedException e) {
```

Comment:
- Shouldn't this be caught only for `lock.tryLock`?
- Does this mean that `addDDF` will successfully return when the thread is interrupted? Isn't that a bad thing?